sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/PushDownUtils.scala

```diff
@@ -65,7 +65,7 @@ object PushDownUtils extends PredicateHelper {
         val postScanFilters = r.pushFilters(translatedFilters.toArray).map { filter =>
           DataSourceStrategy.rebuildExpressionFromFilter(filter, translatedFilterToExpr)
         }
-        (Left(r.pushedFilters()), (untranslatableExprs ++ postScanFilters).toSeq)
+        (Left(r.pushedFilters()), (postScanFilters ++ untranslatableExprs).toSeq)
```

Comment:
This order switching makes sense to me. I think the translated filters (`postScanFilters`) are simple filters that can be evaluated faster, while the untranslated filters are normally complicated filters that take more time to evaluate, so we want to evaluate the `postScanFilters` filters first.

Reply:
cc @cloud-fan

Reply:
> This order switching makes sense to me. I think the translated filters (`postScanFilters`) are simple filters that can be evaluated faster, while the untranslated filters are normally complicated filters that take more time to evaluate, so we want to evaluate the `postScanFilters` filters first.

Thank you for your reply. That's exactly what I was thinking

Reply:
I think this simple heuristic should be safe, although it can't optimize all the cases but it won't make things worse.

Reply:
Can we add a comment here to state that `untranslatableExprs` needs to be on the right side and also briefly explain the reason?